contracts/exchange/TokenPriceProvider.sol

```diff
@@ -36,17 +20,39 @@ contract TokenPriceProvider {
      * @param _token the ERC20 token contract
      * @return the ether value (in wei) of _amount tokens with contract _token
      */
-    function getEtherValue(uint256 _amount, address _token) public view returns (uint256) {
+    function getEtherValue(uint256 _amount, address _token) external view returns (uint256) {
         uint256 decimals = ERC20(_token).decimals();
         uint256 price = cachedPrices[_token];
         return price.mul(_amount).div(10**decimals);
     }
 
     //
-    // Internal
+    // The following is added to be backward-compatible with Argent's old backend
     //
 
-    function kyberNetwork() internal view returns (KyberNetwork) {
-        return KyberNetwork(KYBER_NETWORK_ADDRESS);
+    // Mock token address for ETH
+    address constant internal ETH_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    // Address of the KyberNetwork contract
+    KyberNetwork public kyberNetwork;
+
+    constructor(KyberNetwork _kyberNetwork) public {
+        kyberNetwork = _kyberNetwork;
+    }
+
+    function setKyberNetwork(KyberNetwork _kyberNetwork) external onlyManager {
+        kyberNetwork = _kyberNetwork;
+    }
+
+    function syncPrice(ERC20 token) public {
+        require(address(kyberNetwork) != address(0), "Kyber sync is disabled");
+        uint256 expectedRate;
+        (expectedRate,) = kyberNetwork.getExpectedRate(token, ERC20(ETH_TOKEN_ADDRESS), 10000);
+        cachedPrices[address(token)] = expectedRate;
+    }
+
+    function syncPriceForTokenList(ERC20[] memory tokens) public {
```

Comment:
Can we have the logic of `syncPrice` in `syncPriceForTokenList` and add the require check in `syncPriceForTokenList`? I know it duplicates the code but it will save some gas by avoiding checking the value of `kyberNetwork` for every token.

Reply:
Done. But maybe `syncPrice()` and `setPrice()` could both be `internal` ? @petar-argent do you ever call `syncPrice()` in addition to `syncPriceForTokenList()`?